min.py

```diff
@@ -135,6 +136,73 @@ def get_min_scipy_version(min_python, min_numpy):
     return df.SciPy_version
 
 
+def scan_files_for_minimum_versions(files, package_versions):
+    """
+    Find the current listed versions of packages that align with
+    the specified minimum versions and scan various
+    types of files.
+
+    Args:
+        files (list): A list of file paths to be scanned.
+        package_versions (dict): A dictionary where keys are package
+        names and values are minimum version strings.
+
+    Returns:
+        dict: A dictionary containing package names
+              as keys and lists of tuples as values.
+              Each tuple contains file path, line number,
+              and the outdated version.
+
+    """
+
+    def parse_version(version):
+        """
+        Parse the version string into a tuple for comparison.
+
+        Args:
+            version (str): The version string to parse.
+
+        Returns:
+            tuple: A tuple of integers representing the version.
+
+        """
+        version_parts = version.split(".")
+
+        return tuple(map(int, version_parts))
+
+    # Instantiating a dictionary of lists for storing
+    # any hits on minimum version mismatches while scanning
+    matching_files = {package: [] for package in package_versions}
+
+    # Iteratively scanning files input for minimum version mismatches
+    for file_path in files:
```

Comment:
Having things so deeply nested is concerning and it likely means that you are trying to do too much inside of this function. I think it might be a bit cleaner to, say, have a function called:

```
def find_pkg_mismatches(fnames, pkg, version):
```

So this function only checks for ONE specific `pkg` and `version` pair amongst the set of `fnames`. Then, you would have a separate `for-loop` that looks something like:

```
for pkg, version in package_versions:
    find_pkg_mismatches(fnames, pkg, version)
```

This way, just by looking at this `for-loop`, you know immediately what is happening and the `find_pkg_mismatch` function has a single responsibility (i.e., it doesn't look for all packages and all versions at once). Compact, "single responsibility" functions like this are usually easier to maintain in the longer term

Reply:
Ahhh yeah that makes sense - I'm on it!

I'll try and break this out further to handle single responsibilities!

Reply:
Thank you for being open to feedback!

Reply:
@seanlaw Of course! I truly appreciate the direction, I can't grow without it!

As an FYI I'm hoping to have the new PR by end of week!

Reply:
@seanlaw hey Sean - hope you're having a great one! 

Two clarifying questions...

- Would a dataframe work for the data structure to return regarding all mismatches noted? (or would it be most optimal to use a list or dictionary)
- Are there any opinions from the Stumpy team about default variables with functions? (the way I re-wrote the above seems to require me to have some default variables - but I'd like to stay in line with the Stumpy preferred approach)

Reply:
> Would a dataframe work for the data structure to return regarding all mismatches noted? (or would it be most optimal to use a list or dictionary)

If it's not too painful, I'd like to use `stdlib` if possible but feel free to show your use of `pandas` and we can discuss as `pandas` is already imported. Note that `pandas` is technically NOT a `stumpy` dependency and that it is only there for development purposes only.

> Are there any opinions from the Stumpy team about default variables with functions? (the way I re-wrote the above seems to require me to have some default variables - but I'd like to stay in line with the Stumpy preferred approach)

I'm not sure I understand. I don't see any files/code committed in the current/latest commit